### DISCUSSION TITLE
Unable to change logo, only alt text shown

### DISCUSSION THREAD

**Discussion by confused-llama** [apache/superset#27084] [Q&A / Help]
Apologies in advance if this is somewhat silly. 
I am attempting to customize the look of my installation a little and was attemting to use my custom logo that I have saved in the folder I cloned from git ( `home/user/superset/superset-frontend/src/assets/images` )which I use to spin up my instance with `sudo docker compose -f docker-compose-non-dev.yml up`
I have edited the `superset_config.py` file to include my app name and the new logo.
```
APP_NAME = "My Analytics"
APP_ICON = "/static/assets/images/mylogo52.png"
```
But the image never loads when the instance is spun up again also when using `--build` and only the alt text "My Analytics" is shown. 

I have also just tried replacing the `superset-logo-horiz.png` in the src/assets/images with a file with same name, but it loads the default logo if nothing is set in the `superset_config.py`. This is making me think the installation is pulling the image assets from somwhere else. But where?

I know I'm kinda in over my head here, but I'd love to get it working and would greatly appreciate any help and tips! Thank you!

**Comment by bryanjknight** [comment]
I found a way around this by creating a middleware to capture requests for custom content and serve it locally:

```
class LocalAssetMiddleware:
    """
    LocalAssetMiddleware addresses issue where local assets are not served by superset
    """

    def __init__(self, app):
        self.app = app

    def __call__(self, environ, start_response):
        request = Request(environ)
        if request.method == "GET" and request.path.startswith(
            "/static/assets/images/custom"
        ):
            local_assets_path = f"{config_file_path}/.."
            requested_asset = f"{local_assets_path}/{request.path}"

            # is this a file that we have access to?
            if os.path.exists(requested_asset):
                content_type = get_content_type_by_extension(
                    requested_asset.split(".")[-1]
                )
                content = None
                with open(requested_asset, "rb") as f:
                    content = f.read()
                    start_response("200 OK", [("Content-Type", content_type)])

                # print the data type of data
                return [content]

            return self.app(environ, start_response)

        return self.app(environ, start_response)
```
